crates/state-asm/src/analyse_reads.rs

```diff
@@ -0,0 +1,63 @@
+use crate::StateRead as Op;
+use bitflags::bitflags;
+
+/// Flags to indicate the type of state read operations.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct StateReadFlags(u32);
+
+bitflags! {
+    impl StateReadFlags: u32 {
+        /// No state read operations.
+        const None = 0b00000000;
+        /// [´Op::KeyRange´] state read operations´
+        const Range = 0b00000001;
```

Comment:
It's a little confusing that these are 8bits but the type is u32?
Maybe it would be clearer to do:
```suggestion
        const Range = 1 << 0;
```